framework/src/problems/FEProblemBase.C

```diff
@@ -3709,6 +3709,65 @@ FEProblemBase::projectInitialConditionOnCustomRange(ConstElemRange & elem_range,
   _aux->solution().localize(*_aux->sys().current_local_solution, _aux->dofMap().get_send_list());
 }
 
+void
+FEProblemBase::projectInitialConditionOnCustomRangeForSpecificVars(
+    ConstElemRange & elem_range,
+    ConstBndNodeRange & bnd_nodes,
+    const std::set<std::string> & target_var_names)
+{
+  ComputeInitialConditionThread cic(*this, target_var_names);
+  Threads::parallel_reduce(elem_range, cic);
+
+  // Need to close the solution vector here so that boundary ICs take precendence
+  for (auto & nl : _nl)
+    nl->solution().close();
+  _aux->solution().close();
+
+  ComputeBoundaryInitialConditionThread cbic(*this, target_var_names);
+  Threads::parallel_reduce(bnd_nodes, cbic);
+
+  for (auto & nl : _nl)
+    nl->solution().close();
+  _aux->solution().close();
+
+  // Also, load values into the SCALAR dofs
+  // Note: We assume that all SCALAR dofs are on the
+  // processor with highest ID
+  if (processor_id() == (n_processors() - 1) && _scalar_ics.hasActiveObjects())
+  {
+    const auto & ics = _scalar_ics.getActiveObjects();
+    for (const auto & ic : ics)
+    {
+      MooseVariableScalar & var = ic->variable();
+
+      if (!target_var_names.empty() && !target_var_names.count(var.name()))
+        continue;
+
+      var.reinit();
+
+      DenseVector<Number> vals(var.order());
+      ic->compute(vals);
+
+      const unsigned int n_SCALAR_dofs = var.dofIndices().size();
```

Comment:
```suggestion
      const unsigned int n_scalar_dofs = var.dofIndices().size();
```

see variable naming conventions